### PR TITLE
Migrated authentication and registration application from FBV to CBV

### PR DESCRIPTION
**Pull request by amruthasangeeth**
Migrated authentication application from CBV using Django's auth views.
It used function based views for login and logout process earlier.
Registration application is migrated to TemplateView based calls view which
allows administrator and volunteer signup.

Also implemented generating a message when username and password mismatches
and phone number error.

Updated requirements to avoid tests failure, and corrected URLs

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
cc @nida @smarshy.

File: vms/authentication/views.py [hunk 1]
```diff
@@ -1,40 +1,20 @@
...
-    else:
-        return render(request, 'authentication/login.html', {'is_invalid_credentials' : False,})
 
-@login_required
-def logout_process(request):
 
-    logout(request)
-    return HttpResponseRedirect(reverse('home:index'))
+def anonymous_required(func):
+    """
```
Comment (tapasweni-pathak) [inline vms/authentication/views.py]: @amruthasangeeth Why this name?

File: vms/registration/views.py [hunk 2]
```diff
@@ -2,19 +2,25 @@
...
 from registration.phone_validate import validate_phone
+from administrator.models import *
 
 
-def signup_administrator(request):
+class AdministratorSignupView(TemplateView):
     """
-    This method is responsible for diplaying the register user view
+    Administrator and Volunteer signup is implemented as a TemplateView that
+    displays the signup form
```
Comment (tapasweni-pathak) [inline vms/registration/views.py]: @amruthasangeeth Add a full stop at the end.

File: vms/authentication/views.py (hunk 1 above)
Comment (amruthasangeeth) [inline vms/authentication/views.py]: @tapasweni-pathak This function handles access to restrict access to some views only to non logged-in users. That s why this name.

**Pull request merged**